package/src/utils/DBSyncManager.ts

```diff
@@ -38,6 +38,10 @@ export class DBSyncManager {
   static syncStatus = false;
   static listeners: Array<(status: boolean) => void> = [];
   static client: StreamChat | null = null;
+  static connectionChangedListener: { unsubscribe: () => void } | null = null;
+  static initializationConfig: { initialized: boolean; userID?: string | undefined } = {
+    initialized: false,
+  };
```

Comment:
could you map the `setInitialisedDatabaseConfig({ initialised: false, userID });` to this static config

Reply:
Just remember that we should not render the components for now unless the DB has been initialised for the user

Reply:
Yep, that was my initial question in the comment above - I'd like to do this but wasn't sure why we're setting it to `false` and then immediately to `true`

Reply:
it is not immediate right, we do it after init

Reply:
if you render the component without Db not being initialised, the components will qury from db and fail with db not ready error

Reply:
alternatively we could keep a promise for the init to be completed and make sure that all DB queries await the promise

Reply:
When I say immediate what I mean is, we don't wait for `.init()` to finish before marking it as done. I had something like `DBSyncManager.init().then(() => setInitialisedDatabaseConfig({ initialized: true, userID }))` in mind.